src/hooks/useLiveSource.ts

```
@@ -0,0 +1,161 @@
+  const { state, actions } = store;
+  const { liveSourceManagerRef, textureManagerRef, rendererRef, engineModeRef, renderStateRef } = refs;
+  const liveSource = state.media.liveSource;
+
+  // `refs` (the composed bundle) gets a new object identity every App render
+  // even though its individual ref objects are stable — mirror it into a
+  // stable ref so the tick effect below doesn't need `refs` itself in its
+  // dependency array (which would restart the rAF loop on every render, as
+  // `useAnimationLoop`/`useReactiveInput` avoid by depending on individual
+  // refs instead of the whole bundle).
+  const refsRef = useRef(refs);
+  refsRef.current = refs;
```

ESLint

[error] Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

/home/jailuser/git/src/hooks/useLiveSource.ts:37:3
  35 |   // refs instead of the whole bundle).
  36 |   const refsRef = useRef(refs);
> 37 |   refsRef.current = refs;
     |   ^^^^^^^^^^^^^^^ Cannot update ref during render
  38 |
  39 |   const ensureManager = useCallback((): LiveSourceManager => {
  40 |     liveSourceManagerRef.current ??= new LiveSourceManager(() => {

(react-hooks/refs)